pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -751,17 +776,88 @@ func loadBalancerServiceIsProgressing(ic *operatorv1.IngressController, service
 		err := fmt.Errorf("The IngressController scope was changed from %q to %q.", haveScope, wantScope)
 		switch platform.Type {
 		case configv1.AWSPlatformType, configv1.IBMCloudPlatformType:
-			err = fmt.Errorf("%[1]s  To effectuate this change, you must delete the service: `oc -n %[2]s delete svc/%[3]s`; the service load-balancer will then be deprovisioned and a new one created.  This will most likely cause the new load-balancer to have a different host name and IP address from the old one's.  Alternatively, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/%[4]s --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"loadBalancer\":{\"scope\":\"%[5]s\"}}}}'", err.Error(), service.Namespace, service.Name, ic.Name, haveScope)
+			err = fmt.Errorf("%[1]s  To effectuate this change, you must delete the service: `oc -n %[2]s delete svc/%[3]s`; the service load-balancer will then be deprovisioned and a new one created.  This will most likely cause the new load-balancer to have a different host name and IP address from the old one's.  Alternatively, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/%[4]s --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"loadBalancer\":{\"scope\":\"%[5]s\"}}}}'`", err.Error(), service.Namespace, service.Name, ic.Name, haveScope)
 		}
 		errs = append(errs, err)
 	}
 
+	if platform.Type == configv1.AWSPlatformType {
+		var wantSubnets, haveSubnets *operatorv1.AWSSubnets
+		var paramsFieldName string
+		switch getAWSLoadBalancerTypeInStatus(ic) {
+		case operatorv1.AWSNetworkLoadBalancer:
+			if nlbParams := getAWSNetworkLoadBalancerParametersInSpec(ic); nlbParams != nil {
+				wantSubnets = nlbParams.Subnets
+			}
+			if nlbParams := getAWSNetworkLoadBalancerParametersInStatus(ic); nlbParams != nil {
+				haveSubnets = nlbParams.Subnets
+			}
+			paramsFieldName = "networkLoadBalancer"
+		case operatorv1.AWSClassicLoadBalancer:
+			if clbParams := getAWSClassicLoadBalancerParametersInSpec(ic); clbParams != nil {
+				wantSubnets = clbParams.Subnets
+			}
+			if clbParams := getAWSClassicLoadBalancerParametersInStatus(ic); clbParams != nil {
+				haveSubnets = clbParams.Subnets
+			}
+			paramsFieldName = "classicLoadBalancer"
+		}
+		if !compareAWSSubnets(wantSubnets, haveSubnets) {
+			// Generate JSON for the oc patch command as well as "pretty" json
+			// that will be used for a more human-readable error message.
+			haveSubnetsPatchJson := convertAWSSubnetListToPatchJson(haveSubnets, "null", "null")
+			haveSubnetsPrettyJson := convertAWSSubnetListToPatchJson(haveSubnets, "{}", "[]")
+			wantSubnetsPrettyJson := convertAWSSubnetListToPatchJson(wantSubnets, "{}", "[]")
+			err := fmt.Errorf("The IngressController subnets were changed from %q to %q.", haveSubnetsPrettyJson, wantSubnetsPrettyJson)
```

Comment:
Can this be a simple string (similar to the patch command) just to avoid the reset of `err` which may look like an override?

Reply:
or may be add  errs = append(errs, err) here as well?